sys/shell/shell.c

```diff
@@ -256,6 +256,9 @@ static int readline(char *buf, size_t size)
         else {
             *line_buf_ptr++ = c;
             _putchar(c);
+#ifdef MODULE_NEWLIB
+            fflush(stdout);
+#endif
```

Comment:
I think the last three files here (sys/shell/shell.c, drivers/Makefile.include and  doc/doxygen/riot.doxyfile
) are not related to this PR.